### DISCUSSION TITLE
Inconsitence with the region of show_pdf_page()?

### DISCUSSION THREAD

**Discussion by jesuslop** [pymupdf/PyMuPDF-Utilities#131] [Q&A] [answered]
I am trying to create a PDF hyperlink shown by a custom icon, made clickable. So I want to draw a little icon in a big PDF file, want to render the icon vectorially so I'm using show_pdf_page() to render it. The icon is the file [icon.pdf](https://github.com/pymupdf/PyMuPDF-Utilities/files/15110525/icon.pdf). The big PDF file is generated by Chrome in Windows (print->save to pdf). The file is [test.pdf](https://github.com/pymupdf/PyMuPDF-Utilities/files/15110527/test.pdf).

The problem for me is that the icon and the clickable region **are different**, even if specified by the same rectangle. The problem happens for some big PDFs and not in others (attached is an offending one)

Sample code:
```
import fitz 
from packaging import version

ver = fitz.__version__
if version.parse(ver) < version.parse('1.24.2'):
	raise Exception("latest version required")
	
with fitz.open("test.pdf") as doc:
	page = doc[0]
	rect = fitz.Rect(100, 100, 200, 200)
	with fitz.open("icon.pdf") as icon:
		page.show_pdf_page(rect, icon)
	
	link={}
	link["kind"] = fitz.LINK_URI
	link["uri"] = "http://github.com"
	link["from"] = rect	
	page.insert_link(link)

	doc.save("result.pdf")
```

The result is [result.pdf](https://github.com/pymupdf/PyMuPDF-Utilities/files/15110526/result.pdf).

One can see that the clickable region (visible by a hand mouse cursor when hovering over it) is outside the icon (is about under the title).

**Comment by JorjMcKie** [comment, marked as answer]
There is a problem with automatically recognizing whether a page must be cleaned before inserting stuff - a new feature in this version.

So you must use `page.clean_contents()` directly after `page = doc[0]`.
